### PR TITLE
fix(dev): fix wrong rawUrl in the dev command on redirect

### LINKED ISSUE #3609: When invoking function, `event.rawUrl` value is different from production deploy
**Describe the bug**

When using a function via a 200 rewrite, the value of `event.rawUrl` is the function URL. In production, the value is the original request URL.

**To Reproduce**

- Create a function that logs `event.rawUrl`
- Add a rewrite that points to that function URL
- Run with `ntl dev`, observing URL value
- Deploy to production and observe URL value

**Expected behavior**

The CLI and production values should match. I would prefer the CLI's value (i.e. the function target), but changing prod would be a breaking change so it's the CLI that needs to change (and ideally both should provide another header or field that includes the target URL)

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #3609

Fixes an issue where the `rawUrl` was the original URL once a redirect was created to the function

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

image

### CODE REVIEW

**Review by mraerino — APPROVED** [review]
LGTM

**Review by mraerino — DISMISSED** [review]

File: src/lib/functions/server.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+})
+
+test.after((t) => {
+  t.context.zisiStub.restore()
+})
+
+test('should get the url as the `rawUrl` inside the function', async (t) => {
+  await request(app)
+    .get('/hello')
+    // .set('x-netlify-original-pathname', 'orig')
```
Comment (eduardoboucas) [inline src/lib/functions/server.test.js:53]: Did you mean to leave this here?

**Review by eduardoboucas — APPROVED** [review]
🚀

File: src/lib/functions/server.test.js (hunk 1 above)
Comment (lukasholzer) [reply src/lib/functions/server.test.js:53]: 😬 oh no this one slipped through :D

Comment (lukasholzer) [reply src/lib/functions/server.test.js]: already updated it ;)

Comment (eduardoboucas) [reply src/lib/functions/server.test.js]: Awesome, thanks!

**Review by eduardoboucas — APPROVED** [review]

**Pull request merged**